### PR TITLE
feat: shape our TF module scaffolding

### PR DESCRIPTION
**Pull request by afiune**
This PR is shaping how our Terraform Module Scaffolding will look like! 🙌🏽

tenor-47614410

The idea of this repository is to have an easy way to create new Terraform Modules
for anyone at Lacework.
 
Signed-off-by: Salim Afiune Maya <afiune@lacework.net>

### CODE REVIEW

**Review by scottford-lw — APPROVED** [review]

**Comment by afiune** [comment]
I found this document that explains how to mirror a repo https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/duplicating-a-repository

Though we can't because we have branch protection
```
➜  terraform-module-scaffolding git:(afiune/scaffolding) git push --mirror  https://github.com/lacework/terraform-aws-ecs-agent.git
Enumerating objects: 25, done.
Counting objects: 100% (25/25), done.
Delta compression using up to 12 threads
Compressing objects: 100% (21/21), done.
Writing objects: 100% (25/25), 8.29 KiB | 2.76 MiB/s, done.
Total 25 (delta 1), reused 3 (delta 0)
remote: Resolving deltas: 100% (1/1), done.
remote: error: GH006: Protected branch update failed for refs/heads/main.
remote: error: Cannot force-push to this protected branch
To ssh://github.com/lacework/terraform-aws-ecs-agent.git
 - [deleted]         alannix-lw/initial
 * [new branch]      afiune/scaffolding -> afiune/scaffolding
 * [new branch]      origin/HEAD -> origin/HEAD
 * [new branch]      origin/afiune/scaffolding -> origin/afiune/scaffolding
 * [new branch]      origin/main -> origin/main
 ! [remote rejected] main -> main (protected branch hook declined)
error: failed to push some refs to 'ssh://git@github.com/lacework/terraform-aws-ecs-agent.git'
```

**Comment by afiune** [comment]
Also, I messed up https://github.com/lacework/terraform-aws-ecs-agent/pull/1 🤦🏽‍♂️ - Apologies @alannix-lw 😅

**Comment by afiune** [comment]
This looks much better: 

https://github.com/lacework/terraform-aws-ecs-agent/pull/2

tenor-136445774

**Review by scottford-lw — APPROVED** [review]
👍

File: examples/default/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    lacework = {
+      source = "lacework/lacework"
+    }
+  }
+}
+
+provider "lacework" {}
+
+module "lacework_module" {
+  source  = "lacework/<PROVIDER>/<NAME>"
```
Comment (alannix-lw) [inline examples/default/README.md]:
```suggestion
  source  = "lacework/<NAME>/<PROVIDER>"
```

**Review by alannix-lw — CHANGES_REQUESTED** [review]

File: scripts/release_helpers.sh [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    esac
+    VERSION=$(cat VERSION)
+    log "version bumped from $latest_version to v$VERSION"
+  else
+    log "skipping version bump. Already bumped to v$VERSION"
+    return
+  fi
+
+  log "commiting and pushing the version bump to github"
+  git config --global user.email $git_email
```
Comment (afiune) [inline scripts/release_helpers.sh]:
```suggestion
  git config --global user.email $git_email
```

File: scripts/release_helpers.sh [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+  update_changelog
+  push_release
+  open_pull_request
+}
+
+push_release() {
+  log "commiting and pushing the release to github"
+  _version_no_tag=$(echo $VERSION | awk -F. '{printf("%d.%d.%d", $1, $2, $3)}')
+  if [ "$CI" != "" ]; then
+    git config --global user.email $git_email
```
Comment (afiune) [inline scripts/release_helpers.sh]:
```suggestion
    git config --global user.email $git_email
```

**Pull request merged**